File: Learn Python the Hardway/exercise18/ex18.py
# EXERCISE 18 - Kiswanto_D

# scripts with argv                      
def print_two(*args):
    arg1, arg2 = args
    print "arg1: %r, arg2: %r" % (arg1, arg2)

# *args is actually pointless, we can just do this
def print_two_again(arg1, arg2):
    print "arg1: %r, arg2: %r" % (arg1, arg2)

# one argument
def print_one(arg1):
    print "arg1: %r" % arg1

# no arguments
def print_none():
    print "I got nothin'."


print_two("Dede Kiswanto","Si Keren")
print_two_again("Kiswanto","D")
print_one("First!")
print_none()

File: RubyTheHardway/ex18.rb
def print_two(*args, print_this)
    arg1, arg2 = args
    puts "arg1 : #{arg1}, arg2 : #{arg2}" + " print this : #{print_this}"
end

def print_two_again(arg1, arg2 = "DEFAULT") 
    puts "arg1 : #{arg1}, arg2 : #{arg2}" 
end

def print_none()
    puts "i got nothin."
end

print_two( "Shaw")
print_two_again("Zed")
print_none()

